tests/sentry/deletions/test_group.py

```diff
@@ -31,93 +31,102 @@
 
 
 class DeleteGroupTest(TestCase, SnubaTestCase):
-    def setUp(self) -> None:
-        super().setUp()
-        one_minute = before_now(minutes=1).isoformat()
-        group1_data = {"timestamp": one_minute, "fingerprint": ["group1"]}
-        group2_data = {"timestamp": one_minute, "fingerprint": ["group2"]}
-
-        # Group 1 events
-        self.event = self.store_event(
-            data=group1_data | {"tags": {"foo": "bar"}}, project_id=self.project.id
-        )
-        self.event_id = self.event.event_id
-        self.node_id = Event.generate_node_id(self.project.id, self.event_id)
-        group = self.event.group
-        self.event2 = self.store_event(data=group1_data, project_id=self.project.id)
-        self.node_id2 = Event.generate_node_id(self.project.id, self.event2.event_id)
 
-        # Group 2 event
-        self.keep_event = self.store_event(data=group2_data, project_id=self.project.id)
-        self.keep_node_id = Event.generate_node_id(self.project.id, self.keep_event.event_id)
+    def _generate_data(self, fingerprint: str | None = None) -> dict[str, Any]:
+        return {
+            "fingerprint": [fingerprint or uuid4().hex],
+            "timestamp": before_now(minutes=1).isoformat(),
+        }
+
+    def _get_node_id(self, event: Event) -> str:
+        return Event.generate_node_id(event.project_id, event.event_id)
 
+    def _create_event_with_many_group_children(self) -> Event:
```

Comment:
Only one event required all the extra models:
https://github.com/getsentry/sentry/blob/be7268d66e3d8404572f591d793f7a247d60cba2/tests/sentry/deletions/test_group.py#L54-L69